src/EventListener/HttpCacheListener.php

```diff
@@ -190,7 +190,7 @@ private function convertToSecondsIfNeeded($time)
         if (!is_numeric($time)) {
             $now = microtime(true);
 
-            $time = ceil(strtotime($time, $now) - $now);
+            $time = ceil(strtotime($time, round($now)) - $now);
```

Comment:
Let's not change the logic here. In fact one of the failing tests seems related to your change.

```suggestion
            $time = ceil(strtotime($time, (int) $now) - $now);
```